### PR TITLE
implement filter2D

### PR DESCRIPTION
**Pull request by edgarriba**
implements `filter2D` function similar to OpenCV 
: https://docs.opencv.org/master/d4/d86/group__imgproc__filter.html#ga27c049795ce870216ddfb366086b5a04

### CODE REVIEW

**Comment by edgarriba** [comment]
@ducha-aiki can you review ?

File: kornia/filters/laplacian.py [hunk 1]
```diff
@@ -91,6 +92,9 @@ class Laplacian(nn.Module):
 
     Arguments:
         kernel_size (int): the size of the kernel.
+        borde_type (str): the padding mode to be applied before convolving.
+          The expected modes are: ``'constant'``, ``'reflect'``,
+          ``'replicate'`` or ``'circular'``. Default: ``'reflect'``.
```
Comment (ducha-aiki) [inline kornia/filters/laplacian.py]: >Default: ``'reflect'`` 
Contradicts L114 : , border_type: str = 'replicate'

File: kornia/geometry/transform/pyramid.py [hunk 2]
```diff
@@ -12,20 +14,22 @@
...
 
 
 class PyrDown(nn.Module):
     r"""Blurs a tensor and downsamples it.
 
     Args:
-        input (torch.Tensor): the tensor to be downsampled.
+        borde_type (str): the padding mode to be applied before convolving.
+          The expected modes are: ``'constant'``, ``'reflect'``,
+          ``'replicate'`` or ``'circular'``. Default: ``'reflect'``.
```
Comment (ducha-aiki) [inline kornia/geometry/transform/pyramid.py]: same as Laplacian: "reflect" -> "replicate"

File: kornia/geometry/transform/pyramid.py [hunk 3]
```diff
@@ -68,7 +68,9 @@ class PyrUp(nn.Module):
     r"""Upsamples a tensor and then blurs it.
 
     Args:
-        input (torch.Tensor): the tensor to be upsampled.
+        borde_type (str): the padding mode to be applied before convolving.
+          The expected modes are: ``'constant'``, ``'reflect'``,
+          ``'replicate'`` or ``'circular'``. Default: ``'reflect'``.
```
Comment (ducha-aiki) [inline kornia/geometry/transform/pyramid.py]: same as Laplacian: "reflect" -> "replicate"

**Pull request merged**